capport_core/src/task/requests/http.rs

```diff
@@ -233,4 +319,58 @@ url_column: url
             assert_eq!(actual, expected);
         }
     }
+
+    #[test]
+    fn valid_single_req() {
+        let players = [8478401, 8478402];
+        let teams = ["EDM", "TOR"];
+
+        let server = MockServer::start();
+        let mock: Mock = server.mock(|when, then| {
+            when.method(GET)
+                .path("/v1/meta")
+                .query_param(
+                    "players",
+                    players.iter().map(|n| n.to_string()).collect::<Vec<String>>().join(","),
+                )
+                .query_param("teams", teams.join(","));
+            then.status(200)
+                .header("content-type", "application/json")
+                .body(DummyData::meta_info());
+        });
+
+        let ctx = Arc::new(DefaultContext::default());
+        ctx.insert_result("PLAYERS_DF", df!["players" => players].unwrap().lazy())
+            .unwrap();
+
+        ctx.insert_result("TEAMS_DF", df!["teams" => teams].unwrap().lazy())
+            .unwrap();
+
+        let template = server.url("/v1/meta?players={}&teams={}");
```

Comment:
with the other changes this should become
```suggestion
        let template = server.url("/v1/meta");
```

Reply:
https://api.nhle.com/stats/rest/en/shiftcharts?cayenneExp=gameId%20in%20(2023020573,2024020198)

but this doesn't really work for the `in (a, b, c)` syntax.

If the issue is generating the string N times for N columns, I don't think N will be very big, and actually we can probably avoid regenerating the whole string for each column (just probably more bug prone, and imo not worth).

wdyt, I don't know how to handle the `in (a, b, c)` syntax, otherwise the change looks good and I can implement

Reply:
i forgot about the `in` syntax, then i think instead of the rename attribute you can have an optional template attribute that just calls `format!("gameId in ({})", param_val)` and then joins them together

Reply:
```
    - label: fetch_meta
      task: http_single_request
      args:
        df_to: META_DF
        template: "https://api-web.nhle.com/v1/meta"
        url_params:
          - df_from: PLAYER_DF
            param_column: player
            template: "players={}"
          - df_from: TEAM_DF
            param_column: team
            template: "teams={}"

    - label: fetch_shift_chart
      task: http_single_request
      args:
        df_to: SHIFT_CHART_DF
        template: "https://api.nhle.com/stats/rest/en/shiftcharts"
        url_params:
          - df_from: GAME_DF
            param_column: game
            template: "cayenneExp=gameId%20in%20({})"
```

ok looks something like this now. if template is not present (not shown here but done in a unit test), then it will default to `<param_column>={}` as the formatter. Default fails to work here because API requires players, while I have named the csv column as player.

sorry had to force push because I realized previous commit I accidentally deleted some other test case (was trying a minimal example of another test case to onboard).